src/main/java/tech/stackable/hadoop/StackableTopologyProvider.java

```diff
@@ -175,144 +162,192 @@ private String getLabel(
       String datanode,
       Map<String, Map<String, String>> podLabels,
       Map<String, Map<String, String>> nodeLabels) {
-    String result = new String();
-
     // The internal structures used by this mapper are all based on IP addresses. Depending on
-    // configuration and network setup
-    // it may (probably will) be possible that the namenode uses hostnames to resolve a datanode to
-    // a topology zone.
-    // To allow this, we resolve every input to an ip address below and use the ip to lookup labels.
+    // configuration and network setup it may (probably will) be possible that the namenode uses
+    // hostnames to resolve a datanode to a topology zone. To allow this, we resolve every input to
+    // an ip address below and use the ip to lookup labels.
     // TODO: this might break with the listener operator, as `pod.status.podips` won't contain
     // external addresses
     //      tracking this in https://github.com/stackabletech/hdfs-topology-provider/issues/2
-    InetAddress address = null;
+    InetAddress address;
     try {
       address = InetAddress.getByName(datanode);
-      LOG.debug("Resolved [" + datanode + "] to [" + address.getHostAddress() + "]");
+      LOG.debug("Resolved [{}] to [{}]", datanode, address.getHostAddress());
       datanode = address.getHostAddress();
     } catch (UnknownHostException e) {
       LOG.warn(
-          "failed to resolve address for ["
-              + datanode
-              + "] - this should not happen, defaulting this node to \"defaultRack\".");
+          "failed to resolve address for [{}]}] - this should not happen, "
+              + "defaulting this node to \"defaultRack\".",
+          datanode);
       return "/defaultRack";
```

Comment:
Maybe it's fairly static, but I think if we include a default in a log message, it should be exactly the same:

```suggestion
      public final static String DEFAULT_VALUE = "/defaultRack";
      LOG.warn(
          "failed to resolve address for [{}] - this should not happen, "
              + "defaulting this node to \"{}\".",
          datanode, DEFAULT_VALUE);
      return DEFAULT_VALUE;
```

There was also a superfluous `}]` which I removed in the above.

Reply:
Or define the default as a constant, since it is used elsewhere.

Reply:
Done - see a0f6a13